### ISSUE TITLE
KohanaPHP for PHP7

### ISSUE THREAD

**Issue by MODUS-IT-DEPRECATED** [kohana/kohanaframework.org#31]
Will be kohanaphp developed for PHP 7?

**Comment by enov** [comment]
I just tried to add PHP 7 to Travis and it [did not build](https://travis-ci.org/kohana/kohana/jobs/76061242) because of the old version of [markdown class](https://github.com/kohana/userguide/blob/3.3/master/vendor/markdown/markdown.php) used in userguide.

Having in mind our limited resources, I am not sure if it is a good idea to support PHP 7 for Kohana v3.3.

However, I am :+1: for supporting PHP 7 for the hopeful newer v3.4 release.

I wonder what others would think.

**Comment by acoulton** [comment]
I agree, PHP7 support in 3.4 would be worth having, it's likely too much effort for 3.3 unless someone wants to volunteer to do it.

**Comment by MODUS-IT-DEPRECATED** [comment]
I agree to put it into 3.4. I can voluntee with my company to work on it (we build web apps based on kohanaphp), but I have problem to pull git repository (I can only clone and work standalone without pushing changes to github).
Mayby it would be a good idea to invite to cooperate at kohana some academic groups?

**Comment by MODUS-IT-DEPRECATED** [comment]
Ok, I forked kohana core, but please tell which core branch i should start developing 3.4 / 3.5 ?

**Comment by acoulton** [comment]
Thanks for volunteering! You'll need to do it one module at a time. I suggest starting with kohana/core and kohana/unittest because the other modules depend on them and won't build till they're working.

Fork each project and start a feat/php7-support branch or something like that based on 3.4/develop.

Probably easiest to start with a single commit adding php7 to the .travis.yml, then start a pull request (back to 3.4/develop) and see if it passes. If not then you can start fixing whatever's broken and/or discuss issues with us on the relevant PR. Some of the modules may "just work".

Once all the modules are green then kohana/kohana should be easy.

Does that make sense?

**Comment by MODUS-IT-DEPRECATED** [comment]
I think, I'll need to create new branch, couse in PHP 7 are to many changes, so changes already made in 3.4 are for php 5.5/5.6 so those also will need to be rewrite in some parts.

**Comment by acoulton** [comment]
Dariusz, yes, you need to start a new feature branch (you should always do that for any new work) but it **must** start from and merge back to our 3.4/develop.

We cannot accept your contribution if it doesn't.

You shouldn't need to redo many of the changes for PHP5.5 that have already been done, since those things haven't changed again for PHP7 AFAIK?

**Comment by wwillisn** [comment]
hello.

i am implement web pages and title solutions in kohana. 
I can voluntee for task in kohana. 
the scalable is wonderfull. i am contribute for implement for php7 system.
